mmaction/models/heads/ssn_head.py

```diff
@@ -0,0 +1,439 @@
+        num_classes (int): Number of classes to be classified.
+        with_context (bool): Whether to predict actioness score with
+            context. Default: True.
+        with_regression (bool): Whether to perform regression or not.
+            Default: True.
+        stpp_stage (tuple): Config of structured temporal pyramid pooling.
+            Default: (1, (1, 2), 1).
+    """
+
+    def __init__(self,
+                 num_classes,
+                 with_context=True,
+                 with_regression=True,
+                 stpp_stage=(1, (1, 2), 1)):
+        super().__init__()
+
+        self.with_context = with_context
+
+        self.activity_score_len = num_classes + 1
+        self.complete_score_len = num_classes
+        self.reg_score_len = num_classes * 2
+        self.with_regression = with_regression
+
+        starting_parts, starting_multiplier = parse_stage_config(stpp_stage[0])
+        course_parts, course_multiplier = parse_stage_config(stpp_stage[1])
+        ending_parts, ending_multiplier = parse_stage_config(stpp_stage[2])
+
+        self.feat_multiplier = (
+            starting_multiplier + course_multiplier + ending_multiplier)
+        if self.with_regression:
+            self.feat_dim = (
+                self.activity_score_len + self.feat_multiplier *
+                (self.complete_score_len + self.reg_score_len))
+        else:
+            self.feat_dim = (
+                self.activity_score_len +
+                self.feat_multiplier * self.complete_score_len)
+        self.stpp_stage = (starting_parts, course_parts, ending_parts)
+
+        if self.with_context:
+            self.activity_slice = slice(
+                0, self.activity_score_len * self.feat_multiplier)
+        else:
+            self.activity_slice = slice(0, self.activity_score_len)
+
+        self.complete_slice = slice(
+            self.activity_slice.stop, self.activity_slice.stop +
+            self.complete_score_len * self.feat_multiplier)
+        self.reg_slice = slice(
+            self.complete_slice.stop, self.complete_slice.stop +
+            self.reg_score_len * self.feat_multiplier)
+
+    def forward(self, x, proposal_ticks, scale_factors):
+        """Defines the computation performed at every call.
+
+        Args:
+            x (torch.Tensor): The input data.
+            proposal_ticks (list): Ticks of proposals to be STPP.
+            scale_factors (list): Ratios of the effective sampling lengths
+                to augmented lengths.
+
+        Returns:
+            tuple[torch.Tensor, torch.Tensor, torch.Tensor]:
+                out_activity_scores (torch.Tensor): Activity scores
+                out_complete_scores (torch.Tensor): Completeness scores.
+                out_reg_scores (torch.Tensor): Regression scores.
+        """
+        assert x.size(1) == self.feat_dim
+        num_ticks = proposal_ticks.size(0)
+
+        out_activity_scores = torch.zeros((num_ticks, self.activity_score_len),
```

Comment:
zeros_like, so that both the dtype and device are correct

Reply:
[torch.zeors_like()](https://pytorch.org/docs/stable/generated/torch.zeros_like.html#torch.zeros_like) seems not suitable for this case.